### ISSUE TITLE
Enhancements

### ISSUE THREAD

**Issue by erikyo** [erikyo/vite-block#1]
From a conversation this morning on slack with @gziolo where he was telling me that actually wp-scripts does (also) this:

- [ ] copying PHP files
- [ ] copying block.json files
- [ ] handling 
   - [ ] CSS processing, including PostCSS, SASS, CSS modules, 
   - [ ] Minification
   - [ ] image 
   - [ ] fonts handling
- [ ] JS minification needs to ensure that i18n helpers get preserved
- [ ] React Fast Refresh integration should work
- [ ] entry points are automatically collected from block.json files
- [ ] users can provide their own config overrides to webpack, PostCSS, cssnano, browserslist, babel 

Some things are already there but I keep it as a checklist of the main features

**Comment by gziolo** [comment]
> users can provide their own config overrides to webpack, PostCSS, cssnano, browserslist, babel

That part might not be needed when using Vite, but it illustrates the level of customization that projects use to fulfill their use cases.

**Comment by erikyo** [comment]
You are right, though. Indeed, would be cool to provide the possibility to configure the bundle. Vite has a some options, such as whether to use its internal minifier or Terser, whether to use SWC or Rollup, etc. especially the [latter](https://github.com/vitejs/vite-plugin-react-swc) would be interesting

**Comment by erikyo** [comment]
Perhaps the best approach is to create configurable vite/rollup plugins (they use the same standard).

Yesterday, I have read a little bit about, and I must say that also the resolver part is very improvable, I add here some references:

- https://vitejs.dev/guide/api-plugin#universal-hooks
- https://rollupjs.org/plugin-development/
- https://medium.com/@lior_amsalem/write-your-own-vite-plugin-42aceac9887c
- https://javascript.plainenglish.io/make-your-first-vite-plugin-de5a37ad9fa9
